src/Compiler/Checking/CheckExpressions.fs

```diff
@@ -5585,9 +5585,13 @@ and TcExprUndelayed (cenv: cenv) (overallTy: OverallTy) env tpenv (synExpr: SynE
         CallExprHasTypeSink cenv.tcSink (m, env.NameEnv, overallTy.Commit, env.AccessRights)
         TcConstExpr cenv overallTy env m tpenv synConst
     | SynExpr.DotLambda (synExpr, m, trivia) ->
-        if env.NameEnv.eUnqualifiedItems |> Map.containsKey "_arg1"
-        then
+        match env.NameEnv.eUnqualifiedItems |> Map.tryFind "_arg1" with        
+        // Compiler-generated _arg items can have more forms, the real underscore will be 1-character wide
+        | Some (Item.Value(valRef)) when valRef.Range.StartColumn+1 = valRef.Range.EndColumn ->
```

Comment:
This check looks a bit error-prone to me. Can there be a better check like `IsCompilerGenerated` or a field indicating that the value is created from a unnamed/discard pattern?

Reply:
All of the use cases, both positive and negative, are marked as IsCompilerGenerated. I assume this is because when you type a `fun _`, the underscore gets translated to `_arg1`, and because of this translation it is also marked as being compiler generated.
I have spent quite some time trying to find attributes which differentiate the two uses, but did not find any what so ever.

There is no field I found that would separate the "hidden" discards and a user written discard, except for when looking at the source code (or transitively deriving that from the range).


Since this is only a warning, I do not believe this would justify adding a brand new field to all values...